File: Template/Architecture/VIPER.xctemplate/___FILEBASENAME___/User Interface/Presenter/___FILEBASENAME___Presenter.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  ___COPYRIGHT___
//

#import "___VARIABLE_productName:identifier___Presenter.h"
#import "___VARIABLE_productName:identifier___WireframePort.h"
#import "___VARIABLE_productName:identifier___UserInterfacePort.h"
#import "___VARIABLE_productName:identifier___InteractorPort.h"
//#import "ReactiveCocoa.h"


#define Interactor XFConvertInteractorToType(id<___VARIABLE_productName:identifier___InteractorPort>)
#define Interface XFConvertUserInterfaceToType(id<___VARIABLE_productName:identifier___UserInterfacePort>)
#define Routing XFConvertRoutingToType(id<___VARIABLE_productName:identifier___WireFramePort>)

@interface ___VARIABLE_productName:identifier___Presenter ()

@end

@implementation ___VARIABLE_productName:identifier___Presenter

#pragma mark - lifeCycle
// 绑定视图层后调用
- (void)viewDidLoad
{
    // 解构URL参数
//    NSInteger userID = self.URLParams[@"id"];
}

// 初始化视图数据
- (void)initRenderView
{
    // 填充绑定的ViewData
    //self.viewData = [Interactor fetchData];
}

// 初始化命令
- (void)initCommand
{
    /*XF_CEXE_Begin
    // 当命令触发时执行代码
    XF_CEXE_(self.command, {
        // TODO
    })*/
}

// 接收到组件返回数据
- (void)onNewIntent:(id)intentData
{
    
}

// 注册MVx通知
- (void)registerMVxNotifactions
{
    // 注册MVx构架通知
//    XF_RegisterMVxNotis_(XFUserDidLoginNotifaction)
}


// 接受到MVx构架通知或XFLegoVIPER模块的事件
- (void)receiveComponentEventName:(NSString *)eventName intentData:(id)intentData
{
    // 匹配对应通知
    /*XF_EventIs_(XFUserDidLoginNotifaction, {
        // TODO
    })*/
}

#pragma mark - DoAction



#pragma mark - ValidData


@end

File: Template/Architecture/VIPER@Swift.xctemplate/___FILEBASENAME___/User Interface/Presenter/___FILEBASENAME___Presenter.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  ___COPYRIGHT___
//

import UIKit
import XFLegoVIPER

private func Interactor(_ presenter: XFPresenter) ->___VARIABLE_productName:identifier___InteractorPort? { return presenter.interactor as? ___VARIABLE_productName:identifier___InteractorPort }
private func UserInterface(_ presenter: XFPresenter) ->___VARIABLE_productName:identifier___UserInterfacePort? { return presenter.userInterface as? ___VARIABLE_productName:identifier___UserInterfacePort }
private func Routing(_ presenter: XFPresenter) ->___VARIABLE_productName:identifier___WireFramePort? { return presenter.routing as? ___VARIABLE_productName:identifier___WireFramePort }

class ___VARIABLE_productName:identifier___Presenter: XFPresenter {
    
    override func viewDidLoad() {
        //        print("当前组件名: \(XFComponentReflect.componentName(forComponent: self))")
        //        print("接收到了意图数据：\(self.componentData!)")
    }
    
}

//MARK:- Lego Life Cycle
extension ___VARIABLE_productName:identifier___Presenter {
    // 初始化视图数据
    override func initRenderView() {
        //        Interactor(self)?.fetchData()
    }
    
    // 绑定命令
    override func initCommand() {
        
    }
    
    // 注册通知
    override func registerMVxNotifactions() {
        
    }
    
    // 接收到组件返回数据
    override func onNewIntent(_ intentData: Any) {
        
    }
    
    // 接受到MVx构架通知或组件的事件
    override func receiveComponentEventName(_ eventName: String, intentData: Any?) {
        
    }
}

//MARK:- Action
extension ___VARIABLE_productName:identifier___Presenter : ___VARIABLE_productName:identifier___EventHandlerPort {
    
}

//MARK:- ValidData
extension ___VARIABLE_productName:identifier___Presenter {
    
}
